### ISSUE TITLE
feat(model): speculative candidate tree KV cache recycling and branch demotion

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12529]
# feat(model): speculative candidate tree KV cache recycling and branch demotion

```routing
lane: internal/model
paths: ["internal/model/speculative_branch_cache.go", "internal/model/speculative_branch_cache_test.go"]
expected_steps: 3
```

## Value
- Centrality: Core
- P1: advanced - recycles verified draft KV activations across speculative branch iterations
- P2: advanced - branch cache lookup overhead is sub-5µs, far less than re-evaluating attention layers
- P3: preserved - strictly bounds memory capacity with LRU eviction and enforces causal mask validity
- P4: advanced - produces verified performance receipts in internal/model speculative test suites
- For: high-acceptance tree-speculative decoding and multi-branch agent exploration
- Problem: in internal/model/speculative.go parallelVerifyTree unaccepted candidate branches are destroyed immediately
- Today: 100% of the computed KV activations of rejected branches are purged via evictKV
- Better because: implements SpeculativeBranchCache to recycle unaccepted branch KV buffers for subsequent steps

## Parent context
Native model speculative decoding engine and multi-branch proposal verification.

## Current state
In `internal/model/speculative.go` (`parallelVerifyTree`), when a candidate tree of draft tokens is verified, only the single winning path is committed. The rejected branch tokens are immediately rolled back and purged from the KV cache via `evictKV`. All unaccepted candidate branches in speculative verification are immediately destroyed, wasting 100% of the computed KV activations of rejected branches.

## Why this is next
Iterative multi-agent workflows and tree-based speculation repeatedly revisit related candidate paths; recycling rejected branches avoids recomputation.

## Working spine
Candidate tree verify -> winning path commits -> rejected branch KV buffers demoted to LRU cache -> zero-recompute reuse on match.

## Core through-line
1. Implement `SpeculativeBranchCache` in `internal/model/speculative_branch_cache.go`.
2. Modify `parallelVerifyTree` to preserve unaccepted branch KV buffers into the branch cache before rollback.
3. Add probe and reclaim methods to check for cached branch activations during draft proposal generation.
4. Maintain causal mask invariants and numerical parity with uncached baseline verification.
5. Add unit and benchmark tests verifying zero leaks and >= 40% reduction in draft forward compute on repetitive branching.

## Gold-plating boundary
Strictly excluded:
- Changing target model architecture weights or vocabularies.
- Modifying MTP drafting head architectures.

## Witness
```bash
go test -v ./internal/model -run "TestSpeculativeBranchCache_ReclaimAndParity" -count=1
```

## Acceptance criteria
- [ ] 1. `SpeculativeBranchCache` is implemented in `internal/model/speculative_branch_cache.go`.
- [ ] 2. `parallelVerifyTree` preserves unaccepted branch KV buffers into the branch cache before rollback.
- [ ] 3. Draft generation reclaims cached branch activations without numerical divergence.

## Done condition
Unit test `TestSpeculativeBranchCache_ReclaimAndParity` passes with >= 40% compute reduction on repetitive branching and zero leaks.

## Likely files
- `internal/model/speculative_branch_cache.go`
- `internal/model/speculative_branch_cache_test.go`

**Comment by anthony-chaudhary** [comment]
Landed in commit 3bdcd70db822 and pushed to origin/main in merge 68236055f238:
- Implemented SpeculativeBranchCache in internal/model/speculative_branch_cache.go with bounded LRU eviction and sub-5µs (measured ~15ns) probe and reclaim.
- Modified parallelVerifyTree in internal/model/speculative.go to preserve unaccepted candidate tree KV buffers and logits before rollback.
- Added CachedBranchProposalGenerator and VerifyTreeWithBranchCache to reclaim cached activations on repetitive branching without numerical divergence.
- Added comprehensive unit tests and benchmarks in internal/model/speculative_branch_cache_test.go; TestSpeculativeBranchCache_ReclaimAndParity passes with 100% forward compute reduction (saving 3/3 tokens on repetitive branching), bit-exact numerical parity (delta=0), and zero memory leaks.
- Hardware qualification: PENDING_HARDWARE (Strix Halo unreachable: ssh probe to strix1 failed: exit status 255; next command: fak-dev amd-strix-probe && fak validate --strix).

**Comment by anthony-chaudhary** [comment]
Resolved in commit 3bdcd70db (already landed on main at 93874f690). Verification witness: `go test -v ./internal/model -run TestSpeculativeBranchCache_ReclaimAndParity -count=1` -> PASS (100.00% compute reduction on repetitive branching, 3 saved / 3 total); full package `go test ./internal/model -count=1` -> ok (140.5s). Issue was already-fixed on current trunk; no new code required.
